engineio/src/asynchronous/async_transports/polling.rs

```diff
@@ -81,31 +93,29 @@ impl AsyncTransport for PollingTransport {
         Ok(())
     }
 
-    fn stream(&self) -> Result<Pin<Box<dyn Stream<Item = Result<Bytes>> + '_>>> {
-        let stream = self
-            .address()
-            .into_stream()
-            .map(|address| match address {
-                Ok(addr) => self
-                    .client
-                    .get(addr)
-                    .send()
-                    .map_err(Error::IncompleteResponseFromReqwest)
-                    .left_future(),
-                Err(err) => async { Err(err) }.right_future(),
-            })
-            .then(|resp| async {
-                match resp.await {
-                    Ok(val) => val
-                        .bytes_stream()
-                        .map_err(Error::IncompleteResponseFromReqwest)
-                        .left_stream(),
-                    Err(err) => once(async { Err(err) }).right_stream(),
+    fn stream(&self) -> Pin<Box<dyn Stream<Item = Result<Bytes>> + '_>> {
```

Comment:
suggestion: consume self and return impl Stream
```suggestion
    fn stream(self) -> impl Stream<Item = Result<Bytes>> + '_> {
```

A transport should be converted into a stream, then no longer referenced as a transport.

We may need to setup some channels in order to allow for sending messages

Reply:
Yes, that's a good idea. We only need to separate sender and receiver in that case, as we still want to call methods on the sender after turning the receiver into a stream. I will look into that.

Reply:
May be worth looking into how the Websocket crate implements .split (as that's effectively what we want to do). 

Either that or setup message channels

Reply:
Yeah that comes with an implementation of `Sink` and `Stream` for a type at the same time... https://docs.rs/futures-util/latest/futures_util/stream/trait.StreamExt.html#method.split

Reply:
I've looked into this, implementing Stream is... difficult - to put it nicely.... I think it is a lot of code to get it running for all transports.

Reply:
Could you impl stream by just wrapping an existing stream?
```rust
struct IImplStreamSinkSplit {
stream: Box<dyn Stream<Item=Bytes>>
sink: Box<dyn Sink<Item=Bytes>>
}
struct PollingTransport {}

impl PollingTransport {
  fn into_stream_sink(self) -> (impl Stream<Item=Bytes>, impl Sink<Item=Bytes>) {
    let stream = try_stream! {};
    let sink = try_sink!{}; // Not sure how sinks work
    }
}

impl From<PollingTransport> for ImplStreamSink {
  fn from(polling: obj) -> Self {
    let (stream, sink) = self.into_stream_sink()
    Self {
      stream: Box::new(stream),
      sink: Box::new(sink)
    }
  }
}


```

Reply:
I implemented this but end up with a deadlock... the thing gets messy as the generator always need mutable access to the sender and receiver. Locking the client in the `try_stream!` macro results in a deadlock... Uff this is tough..

Reply:
Yes I think the current version would be fine if we only call the `stream` method once. However I solved the deadlock issue and I am going to commit the new version now. If it's fine with you we could discuss the advantages / disadvantages of both versions ;)